superset/databases/commands/test_connection.py

```diff
@@ -78,7 +81,24 @@ def run(self) -> None:
             )
             with closing(engine.raw_connection()) as conn:
                 try:
+                    alive = func_timeout(
```

Comment:
can we add a small test for handling the func_timeout throwing and verify the the error payload come back how we want it?

I'm thinking a 408 status code would good ?

Reply:
Ah, good point on the 408. Let me added it and add a test.

Reply:
We already have a 408 exception i'd raise this as well
https://github.com/apache/superset/blob/master/superset/exceptions.py#L118